test/chat_models/chat_ollama_ai_test.exs

```diff
@@ -124,6 +131,245 @@ defmodule ChatModels.ChatOllamaAITest do
       assert system_msg["content"] == "You are a weather man"
       assert user_msg["content"] == "What color is the sky?"
     end
+
+    test "generates a map for an API call with a tool", %{ollama_ai: ollama_ai} do
+      fun =
+        Function.new!(%{
+          name: "give_greeting",
+          description: "Gives a friendly greeting for the given subject",
+          parameters_schema: %{
+            type: "object",
+            properties: %{
+              name: %{
+                type: "string",
+                description: "The subject to greet"
+              }
+            },
+            required: ["name"]
+          },
+          function: fn %{"name" => name} = _arguments, _context -> {:ok, "Hello, #{name}!"} end
+        })
+
+      data = ChatOllamaAI.for_api(ollama_ai, [], [fun])
+
+      assert [%{"function" => _} | _] = data.tools
```

Comment:
One of the main goals with the test is to verify it's formatting the data as the targeted LLM expects it. Not just that it's a map with a "function" key. Also, because of the test setup, you can safely make assumptions about the expected data. For example...

```elixir
assert [%{} = result_data] = data.tools

# now create a result_data assertion
```

We want to test that the `LangChain.Function` data structure is structured how the supported Ollama server expects it to be. What does it expect things to be called? Anything different or unusual?